src/shell/focus/mod.rs

```diff
@@ -115,6 +115,13 @@ impl Shell {
                 .unwrap()
                 .element_for_surface(window)
                 .cloned(),
+            Some(KeyboardFocusTarget::LayerSurface(layer)) => {
+                // Do not focus layer surfaces with KeyboardInteractivity::None
+                if !layer.can_receive_keyboard_focus() {
+                    return;
+                }
+                None
+            }
```

Comment:
Yeah this isn't completely right. We want to avoid setting it as the focused surface for smithay's keyboard abstraction instead. Which means not creating a `KeyboardFocusTarget` in the first place from [`element_under`](https://github.com/pop-os/cosmic-comp/blob/master/src/input/mod.rs#L1982).